File: CodeWars/7kyu/highest-and-lowest/main.js
// My Method 1~ 🦉
function highAndLow(numbers) {
    const number_list = numbers.split(" ").map(function (num) {
        return Number(num);
    });

    console.log(number_list);

    const max_and_min = function (operation) {
        return number_list.reduce(function (prevVal, currVal) {
            return operation === "max"
                ? Math.max(prevVal, currVal)
                : Math.min(prevVal, currVal);
        });
    };

    // prettier-ignore
    const max = max_and_min("max"), min = max_and_min("min");

    return `${max} ${min}`;
}

// Method 2:
function highAndLow_2(numbers) {
    const numberArray = numbers.split(" ");
    console.log(...numberArray);
    return `${Math.max(...numberArray)} ${Math.min(...numberArray)}`;
}

const test_case = highAndLow_2("1 2 3 4 5");
console.log(test_case);

// Tests Reduce and Split
const split = "1 2 3 4 5".split(" ").map(function (num) {
    return Number(num);
});

const reduce = split.reduce(function (a, b) {
    return Math.max(a, b);
});

File: CodeWars/7kyu/highest-and-lowest/main.py
# My Method 1~ 🦉
def high_and_low(numbers):
    list_numbers = list(map(int, numbers.split(" ")))
    return f"{max(list_numbers)} {min(list_numbers)}"


# Method 2:
def high_and_low_2(numbers):
    nums = list(map(int, numbers.split(" ")))
    return " ".join(map(str, (max(nums), min(nums))))


test_case = high_and_low_2("8 3 -5 42 -1 0 0 -9 4 7 4 -4")
print(test_case)
